w3-filecoin.md

```diff
@@ -0,0 +1,658 @@
+
+```mermaid
+sequenceDiagram
+    actor Storefront as <br/><br/>did:web:web3.storage
+    actor Aggregator as <br/><br/>did:key:agg...
+
+    Storefront->>Aggregator: invoke `piece/add`<br>with:`did:web:web3.storage`
+    Note left of Aggregator: Request piece to be included in aggregate
+    activate Aggregator
+    Aggregator-->>Storefront: receipt issued as `queued`
+    Aggregator->>Aggregator: invoke `piece/add`<br>with:`did:key:agg...`
+    deactivate Aggregator
+    Aggregator-->>Storefront: receipt issued <br>with inclusion proof
+```
+
+### Aggregator offers broker an aggregate
+
+When the Aggregator has enough content to fulfill an aggregate (each broker MAY have different requirements), a Filecoin deal for an aggregate MAY be requested by an `aggregate/offer` invocation. Deal negotiations with Filecoin Storage Providers MAY be handled out of band. Broker MUST acknowledge a request by issuing a signed receipt.
+
+Once a Broker successfully receives the offer of an aggregate, the aggregate gets queued for deals with Storage Providers. A receipt is created to proof the transition of the offered aggregate state from `null` into `queued`. It is worth mentioning that if an offer is for an aggregate that is already `queued` or `accepted`, it is a NOP.
+
+This receipt MUST have link to a followup task (using `.fx.join` field) that either succeeds (if the aggregate was added into a deal) or fails (if the aggregate was determined to be invalid) so that its receipt COULD be looked up using it.
+
+After a broker dequeues the aggregate, it will interact with available Filecoin Storage Providers, in order to establish a previously determined (out of band) number of deals. Depending on storage providers availability, as well as the content present in the offer, the aggregate MAY be handled or not. A receipt is created to proof the transition of the aggregate state from `queued` into `accepted` or `rejected`.
+
+> Note: Broker MAY have several intermediate steps and states it transitions through, however those intentionally are not captured by this protocol, because storefront will take no action until success / failure condition is met.
+
+```mermaid
+sequenceDiagram
+    actor Aggregator as <br/><br/>did:key:agg...
+    actor Broker as <br/><br/>did:key:brk...
+
+    Aggregator->>Broker: invoke `aggregate/add`<br>with:`did:key:agg...`
+    Note left of Broker: Request aggregate to be queued for deal proposal
+    activate Broker
+    Broker-->>Aggregator: receipt issued as `queued`
+    Broker->>Broker: invoke `aggregate/add`<br>with:`did:key:brk...`
+    deactivate Broker
+    Broker-->>Aggregator: receipt issued with `accepted` or `rejected`
+```
+
+If the aggregate reaches the `accepted` state, the broker takes care of renewing deals.
+
+The broker MAY request an out of bound signature from the Storefront to validate the terms of a deal.
```

Comment:
This is about Broker asking Storefront to sign the deal... as in the w3-deal spec.

The Broker (now dealer) receives an aggregate offer and when it matches it with SPs, will request a out of bound signature of the deal terms.

We can update in the future to be storefront to delegate to broker capability to submit to broker. But I would keep things simpler for the first iteration so that we can move forward for the MVP